invenio_app_ils/records/views.py

```diff
@@ -9,28 +9,112 @@
 
 from __future__ import absolute_import, print_function
 
-from flask import Blueprint, abort, current_app, request
+from flask import Blueprint, abort, current_app, make_response, request
 from invenio_db import db
 from invenio_files_rest.models import ObjectVersion
 from invenio_files_rest.signals import file_downloaded
 from invenio_records_rest.utils import obj_or_import_string
 from invenio_records_rest.views import pass_record
 from invenio_rest import ContentNegotiatedMethodView
 from invenio_rest.errors import FieldError
+from webargs import fields
+from webargs.flaskparser import use_args
 
 from invenio_app_ils.errors import DocumentRequestError, StatsError
 from invenio_app_ils.permissions import need_permissions
 from invenio_app_ils.pidstore.pids import DOCUMENT_PID_TYPE, \
-    DOCUMENT_REQUEST_PID_TYPE, EITEM_PID_TYPE
+    DOCUMENT_REQUEST_PID_TYPE, EITEM_PID_TYPE, PATRON_PID_TYPE
 from invenio_app_ils.proxies import current_app_ils
+from invenio_app_ils.records.serializers.access import patron_list_serializer
 from invenio_app_ils.signals import record_viewed
 
 
+def create_access_list_blueprint(app):
+    """Create access list blueprint."""
+    blueprint = Blueprint("ils_access_list", __name__, url_prefix="")
+    endpoints = app.config.get("RECORDS_REST_ENDPOINTS", [])
+
+    # Get the list not the document
+    options = endpoints.get(PATRON_PID_TYPE, {})
+    default_media_type = options.get("default_media_type", "")
+    serializers = {"application/json": patron_list_serializer}
+    access_list_view = AccessListResource.as_view(
+        AccessListResource.view_name,
+        serializers=serializers,
+        default_media_type=default_media_type,
+    )
+    blueprint.add_url_rule(
+        "/access/list",
+        view_func=access_list_view,
+        methods=["GET"],
+    )
+    return blueprint
+
+
+class AccessListResource(ContentNegotiatedMethodView):
+    """Access list endpoint."""
+
+    view_name = "access_list"
+
+    @use_args({"q": fields.Str(missing='')})
```

Comment:
What about to define a `q_args` property and use that in the `use_args` decorator? I am thinking that if we do it like that it would be easier for someone to inherit this view and add potentially more args. See also an example [here](https://github.com/inveniosoftware/invenio-accounts/pull/295/files#diff-223576a60bd9ea79383ec3cf2d503ebbR200)